sweep/tx_input_set.go

```diff
@@ -125,3 +174,71 @@ func (t *txInputSet) addPositiveYieldInputs(sweepableInputs []txInput,
 
 	// We managed to add all inputs to the set.
 }
+
+// tryAddWalletInputsIfNeeded retrieves utxos from the wallet and tries adding as
+// many as required to bring the tx output value above the given minimum.
+func (t *txInputSet) tryAddWalletInputsIfNeeded() error {
+	// If we've already reached the dust limit, no action is needed.
+	if t.dustLimitReached() {
+		return nil
+	}
+
+	// Retrieve wallet utxos. Only consider confirmed utxos to prevent
+	// problems around RBF rules for unconfirmed inputs.
+	utxos, err := t.wallet.ListUnspentWitness(1, math.MaxInt32)
```

Comment:
to avoid having the wallet as a field on the input set, the list of utxos can be passed into this method.

Reply:
yes, but then we'd always get utxos, even if the dust limit is already reached. could also move that out then. but not sure if it is better